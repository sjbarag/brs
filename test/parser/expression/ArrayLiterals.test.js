const BrsError = require("../../../lib/Error");
const { Lexeme, BrsTypes, Parser } = require("brs");
const { BrsBoolean, Int32 } = BrsTypes;

const { EOF } = require("../ParserTests");

describe("parser array literals", () => {
    afterEach(() => BrsError.reset());

    describe("empty arrays", () => {
        test("on one line", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.RightSquare, text: "]", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.Newline, text: "\n", line: 4 },
                { kind: Lexeme.Newline, text: "\n", line: 5 },
                { kind: Lexeme.Newline, text: "\n", line: 6 },
                { kind: Lexeme.RightSquare, text: "]", line: 7 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });

    describe("filled arrays", () => {
        test("on one line", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Integer, text: "2", line: 1, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Integer, text: "3", line: 1, literal: new Int32(3) },
                { kind: Lexeme.RightSquare, text: "]", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines with commas", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Integer, text: "2", line: 2, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 2 },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Integer, text: "3", line: 3, literal: new Int32(3) },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.RightSquare, text: "]", line: 4 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines without commas", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Integer, text: "2", line: 2, literal: new Int32(2) },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Integer, text: "3", line: 3, literal: new Int32(3) },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.RightSquare, text: "]", line: 4 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });

    describe("contents", () => {
        it("can contain primitives", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Integer, text: "2", line: 1, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Integer, text: "3", line: 1, literal: new Int32(3) },
                { kind: Lexeme.RightSquare, text: "]", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("can contain other arrays", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                    { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                    { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                    { kind: Lexeme.Comma, text: ",", line: 1 },
                    { kind: Lexeme.Integer, text: "2", line: 1, literal: new Int32(2) },
                    { kind: Lexeme.Comma, text: ",", line: 1 },
                    { kind: Lexeme.Integer, text: "3", line: 1, literal: new Int32(3) },
                    { kind: Lexeme.RightSquare, text: "]", line: 1 },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                    { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                    { kind: Lexeme.Integer, text: "4", line: 1, literal: new Int32(4) },
                    { kind: Lexeme.Comma, text: ",", line: 1 },
                    { kind: Lexeme.Integer, text: "5", line: 1, literal: new Int32(5) },
                    { kind: Lexeme.Comma, text: ",", line: 1 },
                    { kind: Lexeme.Integer, text: "6", line: 1, literal: new Int32(6) },
                    { kind: Lexeme.RightSquare, text: "]", line: 1 },
                { kind: Lexeme.RightSquare, text: "]", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("can contain expressions", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftSquare, text: "[", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Plus, text: "+", line: 1},
                { kind: Lexeme.Integer, text: "2", line: 1, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Not, text: "not", line: 1 },
                { kind: Lexeme.False, text: "false", line: 1, literal: BrsBoolean.False },
                { kind: Lexeme.RightSquare, text: "]", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});
