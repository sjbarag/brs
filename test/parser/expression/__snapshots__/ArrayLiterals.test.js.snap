// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser array literals contents can contain expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        Binary {
          "left": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 1,
            },
          },
          "right": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 2,
            },
          },
          "token": Object {
            "isReserved": false,
            "kind": "Plus",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "+",
          },
          "type": "Binary",
        },
        Unary {
          "operator": Object {
            "isReserved": true,
            "kind": "Not",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "not",
          },
          "right": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": BrsBoolean {
              "kind": 2,
              "value": false,
            },
          },
          "type": "Unary",
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals contents can contain other arrays 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        ArrayLiteral {
          "close": Object {
            "isReserved": false,
            "kind": "RightSquare",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "]",
          },
          "elements": Array [
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 1,
              },
            },
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 2,
              },
            },
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 3,
              },
            },
          ],
          "open": Object {
            "isReserved": false,
            "kind": "LeftSquare",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "[",
          },
          "type": "ArrayLiteral",
        },
        ArrayLiteral {
          "close": Object {
            "isReserved": false,
            "kind": "RightSquare",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "]",
          },
          "elements": Array [
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 4,
              },
            },
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 5,
              },
            },
            Literal {
              "_location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "type": "Literal",
              "value": Int32 {
                "kind": 4,
                "value": 6,
              },
            },
          ],
          "open": Object {
            "isReserved": false,
            "kind": "LeftSquare",
            "literal": undefined,
            "location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "text": "[",
          },
          "type": "ArrayLiteral",
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals contents can contain primitives 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 1,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 2,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 3,
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals empty arrays on multiple lines 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals empty arrays on one line 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals filled arrays on multiple lines with commas 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 1,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 2,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 3,
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals filled arrays on multiple lines without commas 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 1,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 2,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 3,
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;

exports[`parser array literals filled arrays on one line 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": ArrayLiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "]",
      },
      "elements": Array [
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 1,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 2,
          },
        },
        Literal {
          "_location": Object {
            "end": Object {
              "column": -9,
              "line": -9,
            },
            "start": Object {
              "column": -9,
              "line": -9,
            },
          },
          "type": "Literal",
          "value": Int32 {
            "kind": 4,
            "value": 3,
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftSquare",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "[",
      },
      "type": "ArrayLiteral",
    },
  },
]
`;
